Mathlib/RingTheory/Ideal/Span.lean

```diff
@@ -257,3 +257,46 @@ theorem ker_toSpanSingleton_one_sub_eq_span :
   rw [ker_toSpanSingleton_eq_span he.one_sub, sub_sub_cancel]
 
 end IsIdempotentElem
+
+namespace Subideal
+
+open Subtype
+
+variable {A : Type*} [CommSemiring A] {I : Ideal A}
+
+/-- The Galois coinsertion between the map sending a subideal `J ≤ I` to itself,
+  and the mal sending `J ↦ J ⊓ I`. -/
+def galoisCoinsertion :
+    GaloisCoinsertion (fun J : { J : Ideal A // J ≤ I } ↦ (J : Ideal A))
+      (fun J : Ideal A ↦ ⟨J ⊓ I, inf_le_right⟩) :=
+  GaloisCoinsertion.monotoneIntro (fun _ _ h ↦ mk_le_mk.mpr (inf_le_inf_right I h))
+    (fun _ _ h ↦ h) (fun _ ↦ inf_le_left)
+    (fun ⟨J, hJ⟩ ↦ by simp only [ge_iff_le, mk.injEq, inf_eq_left, hJ])
+
+instance : CompleteLattice { J : Ideal A // J ≤ I } :=
```

Comment:
We have [Set.Iic.instCompleteLattice](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/CompleteLatticeIntervals.html#Set.Iic.instCompleteLattice)

Reply:
Thanks, we were not aware of this (and it is not found by `inferInstance` or `apply?`). I have changed the code to use it.